rados/striper/stat.go

```diff
@@ -0,0 +1,58 @@
+//go:build (octopus || pacific || quincy) && ceph_preview
+
+package striper
+
+// #cgo LDFLAGS: -lrados -lradosstriper
+// #include <stdlib.h>
+// #include <radosstriper/libradosstriper.h>
+import "C"
+
+import (
+	"unsafe"
+
+	ts "github.com/ceph/go-ceph/internal/timespec"
+)
+
+// Timespec behaves similarly to C's struct timespec.
+type Timespec ts.Timespec
+
+// StatInfo contains values returned by a Striper's Stat call.
+type StatInfo struct {
+	Size    uint64
+	ModTime Timespec
+}
+
+// Stat returns metadata describing the striped object.
+// This version of Stat uses an older API that does not provide time
+// granularity below a second: the Nsec value of the StatInfo.ModTime field
+// will always be zero.
+//
+// Implements:
+//
+//	int rados_striper_stat(rados_striper_t striper,
+//	                       const char* soid,
+//	                       uint64_t *psize,
+//	                       time_t *pmtime);
+func (s *Striper) Stat(soid string) (StatInfo, error) {
```

Comment:
In order to guarantee that the exported API doesn't differ between versions, wouldn't it be better to have a single common file with all exported symbols, and then call internal implementations from there? I also wonder what our API check scripts do, probably use the version without tags?

Reply:
> wouldn't it be better to have a single common file with all exported symbols

Could you elaborate more on what you mean by this, I'm not following.

> probably use the version without tags?

Yes, that's correct.

Reply:
Sorry if it wasn't clear enough. What I mean is, you have one `stat.go` file that defines the exported types and and functions, and then for instance `stat_legacy.go` and `stat2.go` that both define a _non_-exported function `func (s *Striper) stat(soid string) (StatInfo, error)`. In `stat.go` the function would just be a wrapper:

```go
func (s *Striper) Stat(soid string) (StatInfo, error) {
    return s.stat(soid)
}
```

That way you don't have redundant exported definition and avoid that by accident some exported details differ. (Now you define all the types and functions twice without guarantee that they are identical.)

Reply:
Ah, I see. Thanks for clarifying.  I had even briefly considered something like that. Sure, I can make that change.

Reply:
I remember now why I didn't do it. If I define StatInfo and Timspec in a stat.go with a "shell" Stat function, then I still need to refer to it from the "real" stat2.go and stat_legacy.go. Even if go supports that I don't like the two files having a "circular dependency" on each other. What I can do instead is just move the common bits to stat_info.go which will avoid the "redundant exported definition and avoid that by accident some exported details differ" situation you pointed out.

Reply:
Done. PTAL.